src/charm.py

```diff
@@ -193,27 +227,84 @@ def _get_unit_fqdn(self, unit_name: str) -> str:
         """
         return f"{self.get_unit_hostname(unit_name)}.{self.model.name}.svc.cluster.local"
 
-    def get_secret(self, scope: str, key: str) -> Optional[str]:
+    def get_secret(
+        self,
+        scope: str,
+        key: str,
+        relation: str = PEER,
+    ) -> Optional[str]:
         """Get secret from the secret storage."""
+        if relation not in [PEER, DATABASE_BACKUPS_PEER]:
+            raise RuntimeError("Unknown peer relation")
```

Comment:
reverted to original code in d0c834aed6e0e0f47ae416af3f5ad9d6edb56948 - dont want to change original code too much since it'll soon be replaced with juju secrets